homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,296 @@
+                data_schema=vol.Schema({CONF_USB_SPHERE: vol.In(spheres.keys())}),
+            )
+
+        self.usb_sphere_id = spheres[user_input[CONF_USB_SPHERE]]
+        return self.async_create_new_entry()
+
+    def async_create_new_entry(self) -> FlowResult:
+        """Create a new entry."""
+        return self.async_create_entry(
+            title=f"Account: {self.login_info[CONF_EMAIL]}",
+            data={
+                CONF_EMAIL: self.login_info[CONF_EMAIL],
+                CONF_PASSWORD: self.login_info[CONF_PASSWORD],
+                CONF_USB_PATH: self.usb_path,
+                CONF_USB_SPHERE: self.usb_sphere_id,
+            },
+        )
+
+
+class CrownstoneOptionsFlowHandler(OptionsFlow):
+    """Handle Crownstone options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize Crownstone options."""
+        self.entry = config_entry
+        self.updated_entry_data = config_entry.data.copy()
+        self.spheres: dict[str, str] = {}
+        self.options_input: dict[str, Any] = {}
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage Crownstone options."""
+        manager: CrownstoneEntryManager = self.hass.data[DOMAIN][self.entry.entry_id]
+
+        spheres = {sphere.name: sphere.cloud_id for sphere in manager.cloud.cloud_data}
+        usb_path = self.entry.data.get(CONF_USB_PATH)
+        usb_sphere = self.entry.data.get(CONF_USB_SPHERE)
+
+        options_schema = vol.Schema(
+            {vol.Optional(CONF_USE_USB_OPTION, default=usb_path is not None): bool}
+        )
+        if usb_path is not None and len(spheres) > 1:
+            options_schema = options_schema.extend(
+                {
+                    vol.Optional(
+                        CONF_USB_SPHERE_OPTION,
+                        default=manager.cloud.cloud_data.spheres[usb_sphere].name,
+                    ): vol.In(spheres.keys())
+                }
+            )
+
+        if user_input is not None:
+            if user_input[CONF_USE_USB_OPTION] and usb_path is None:
+                self.options_input = user_input
+                self.spheres = spheres
+                return await self.async_step_usb_config_option()
+            if not user_input[CONF_USE_USB_OPTION] and usb_path is not None:
+                self.updated_entry_data[CONF_USB_PATH] = None
+                self.updated_entry_data[CONF_USB_SPHERE] = None
+                user_input.pop(CONF_USB_SPHERE_OPTION)
+            elif (
+                CONF_USB_SPHERE_OPTION in user_input
+                and spheres[user_input[CONF_USB_SPHERE_OPTION]] != usb_sphere
+            ):
+                sphere_id = spheres[user_input[CONF_USB_SPHERE_OPTION]]
+                user_input[CONF_USB_SPHERE_OPTION] = sphere_id
+                self.updated_entry_data[CONF_USB_SPHERE] = sphere_id
+
+            self.entry.data = MappingProxyType(self.updated_entry_data)
```

Comment:
Don't set this directly, let `async_create_entry` set it.

Reply:
When the flow finishes <https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L1447> will take care of updating the options.  If you have options that will change that are configured via the options flow and the original config flow, you can store them in `options` from the start using https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L1393

Reply:
aha, so `CONF_USB_PATH` and `CONF_USB_SPHERE` should go into the options directly, thanks.